sui-execution/latest/sui-move-natives/src/crypto/sui_state_proof.rs

```diff
@@ -0,0 +1,387 @@
+    /// Base cost for invoking the `sui_state_proof_verify_committee` function
+    pub sui_state_proof_verify_committee_cost_base: InternalGas,
+    /// Base cost for invoking the `sui_state_proof_verify_link_cap` function
+    pub sui_state_proof_verify_link_cap_base: InternalGas,
+    /// Base cost for invoking the `sui_state_proof_verify_transaction` function
+    pub sui_state_proof_verify_transaction_base: InternalGas,
+}
+
+/***************************************************************************************************
+ * native fun sui_state_proof_verify_committee
+ * Implementation of the Move native function `sui_state_proofs::sui_state_proof_verify_committee(commitment_to_centralized_party_secret_key_share: vector<u8>, secret_key_share_encryption_and_proof: vector<u8>, centralized_party_public_key_share_decommitment_and_proofs: vector<u8>): (vector<u8>, vector<u8>, vector<u8>);`
+ *   gas cost: sui_state_proof_verify_committee_cost_base   | base cost for function call and fixed opers
+ **************************************************************************************************/
+pub fn sui_state_proof_verify_committee(
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 2);
+
+
+    // Load the cost parameters from the protocol config
+    let sui_state_proof_cost_params = &context
+        .extensions()
+        .get::<NativesCostTable>()
+        .sui_state_proof_cost_params
+        .clone();
+
+    // Charge the base cost for this operation  
+    native_charge_gas_early_exit!(
+        context,
+        sui_state_proof_cost_params.sui_state_proof_verify_committee_cost_base
+    );
+
+    let cost = context.gas_used();
+
+
+    let checkpoint_summary_bytes = pop_arg!(args, Vec<u8>);
+    let prev_committee_bytes = pop_arg!(args, Vec<u8>);
+
+    let Ok(prev_committee) = bcs::from_bytes::<Committee>(&prev_committee_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_COMMITTEE
+        ));
+    };
+
+    
+    let Ok(checkpoint_summary) = bcs::from_bytes::<CertifiedCheckpointSummary>(&checkpoint_summary_bytes) else {
+        return Ok(NativeResult::err(
+            cost,
+            INVALID_CHECKPOINT_SUMMARY
+        ));
+    };
+
+    match checkpoint_summary.clone().verify(&prev_committee) {
+        Ok((_)) => (),
+        Err(e) => return Ok(NativeResult::err(cost, INVALID_TX)),
+    }
+
+
+    let next_committee_epoch;
+    // Extract the new committee information
+    if let Some(EndOfEpochData {
+        next_epoch_committee,
+        ..
+    }) = &checkpoint_summary.end_of_epoch_data
+    {
+        let next_committee = next_epoch_committee.iter().cloned().collect();
+        next_committee_epoch =
+            Committee::new(checkpoint_summary.epoch().checked_add(1).unwrap(), next_committee);
```

Comment:
```suggestion
        // epochs won't overflow since epoch is a 64-bit number and an epoch happens every few hours.
        next_committee_epoch =
             Committee::new(checkpoint_summary.epoch() + 1, next_committee);
```
There's no point in doing a `checked_add()` if you don't handle the failure (you just `unwrap()`).